src/lib/ActionManager.js

```diff
@@ -0,0 +1,49 @@
+// This is from redux
+function compose(...funcs) {
+ return (...args) => {
+   if (funcs.length === 0) {
+     return args[0];
+   }
+
+   const last = funcs[funcs.length - 1];
+   const rest = funcs.slice(0, -1);
+
+   return rest.reduceRight((composed, f) => f(composed), last(...args));
+ };
+}
+
+module.exports = class ActionManager {
+  constructor(types) {
+    this._types = new Set(types);
+    this.validators = [this.validateStandardForm, this.validateType];
+    this.actions = {};
+    this.defineActions({
+      Action: action => action
```

Comment:
should that be `action => action()`? Or calling the action returns the action? (i.e., `action()()`)?